pkg/virt-operator/resource/generate/components/crds.go

```diff
@@ -637,10 +635,8 @@ func NewKubeVirtCR(namespace string, pullPolicy corev1.PullPolicy, featureGates
 		}
 	}
 
-	if infraReplicas != util.DefaultInfraReplicas {
-		cr.Spec.Infra = &virtv1.ComponentConfig{
-			Replicas: &infraReplicas,
-		}
+	cr.Spec.Infra = &virtv1.ComponentConfig{
+		Replicas: &infraReplicas,
```

Comment:
I'm also confused by this.
[Here](https://github.com/kubevirt/kubevirt/blob/main/tools/resource-generator/resource-generator.go#L61), if `--infraReplicas` is not specified, we pass the `infraReplicas` as a placeholder (which equals `255`).

Why is this needed at all?
Why not do something like:
```go
if infraReplicas != 0 {
    cr.Spec.Infra = &virtv1.ComponentConfig{
        Replicas: &infraReplicas,
    }
}
```

Thanks

Reply:
In upstream, I believe `infraReplicas` will always be `255` here, since we also only use this for templates.
However, it could be used for other purposes. We could sanity check infraReplicas and panic if it's 0, but just adding that `if` might cause confusion...